### PR TITLE
UI grocery page

### PR DESCRIPTION
**Pull request by Brianna-Flo**
## Description

<what this pull request is doing>

- implementing react components on grocery page
- Adjusting Generic List to present grocery departments correctly

<what will be done in later PRs and not included here>

- moving styling to universally apply across components in app

## Milestones
<the milestones or stories/features that this works towards>

- creating a cohesive app ui

## Resources
<links to tutorials, code snippets, inspirations>
<if you took or translated specific code from the internet, please call it out here!>

## Test Plan
<insert images or gifs of feature>
<otherwise, say how code was tested if not UI facing>
<any edge cases you might have specifically tested>
https://docs.google.com/document/d/1u4QuX_0hLM_7-EKXGbdEtpxEErOixgLAIE5qFVQna7s/edit?tab=t.ze4hhc76dlp

### CODE REVIEW

File: frontend/src/components/Ingredient.tsx [hunk 1]
```diff
@@ -51,7 +51,7 @@ const Ingredient: React.FC<GPIngredientProps> = ({
             />
           </Grid>
         )}
-        <Grid xs={presentGroceryCheck ? 3 : 4}>
+        <Grid xs={4}>
```
Comment (nithyad) [inline frontend/src/components/Ingredient.tsx:54]: since they are both 4, lets make a const

**Review by nithyad — APPROVED** [review]
